include/dgl/aten/coo.h

```diff
@@ -138,16 +145,21 @@ struct COOMatrix {
 
   /** @brief Return a copy of this matrix in pinned (page-locked) memory. */
   inline COOMatrix PinMemory() {
-    if (is_pinned) return *this;
-    auto new_coo = COOMatrix(
-        num_rows, num_cols, row.PinMemory(), col.PinMemory(),
-        aten::IsNullArray(data) ? data : data.PinMemory(), row_sorted,
-        col_sorted);
-    CHECK(new_coo.is_pinned)
-        << "An internal DGL error has occured while trying to pin a COO "
-           "matrix. Please file a bug at 'https://github.com/dmlc/dgl/issues' "
-           "with the above stacktrace.";
-    return new_coo;
+    if (!IsEmpty()) {
+      if (is_pinned) return *this;
+      auto new_coo = COOMatrix(
+          num_rows, num_cols, row.PinMemory(), col.PinMemory(),
+          aten::IsNullArray(data) ? data : data.PinMemory(), row_sorted,
+          col_sorted);
+      CHECK(new_coo.is_pinned)
+          << "An internal DGL error has occured while trying to pin a COO "
+             "matrix. Please file a bug at "
+             "'https://github.com/dmlc/dgl/issues' "
+             "with the above stacktrace.";
+      return new_coo;
+    }
+    is_pinned = true;
```

Comment:
`is_pinned` will be set to true even if `IsEmpty() == true`, is it expected?

Reply:
You're right. `is_pinned` should not be set for this pinning method :( 
Thank you!

Reply:
Given the test failures, now, I start thinking I probably should set `is_pinned ` to True even if `IsEmpty() == true`, for both in-place and out-of-place methods. If graph is empty, we should just change the flag in these functions. Then, out-of-place method becomes the same as in-place method as there are no memory movements, but just flag changing.